consensus/hotstuff/verification/combined_verifier.go

```diff
@@ -122,3 +123,65 @@ func (c *CombinedVerifier) VerifyQC(signers flow.IdentityList, sigData []byte, b
 	}
 	return stakingValid, nil
 }
+
+// VerifyConsensusQC verifies the validity of a combined signature on a quorum certificate.
+func (c *CombinedVerifier) VerifyConsensusQC(signers flow.IdentityList, sigData []byte, block *model.Block) error {
+
+	dkg, err := c.committee.DKG(block.BlockID)
+	if err != nil {
+		return fmt.Errorf("could not get dkg: %w", err)
+	}
+
+	aggregatedSigData, err := c.packer.Unpack(block.BlockID, signers, sigData)
+	if err != nil {
+		return fmt.Errorf("could not unpack sig data for block (%v): %w", block.BlockID, sigData)
+	}
+
+	msg := MakeVoteMessage(block.View, block.BlockID)
+	// TODO: verify if batch verification is faster
+
+	// verify the beacon signature first
+	beaconValid, err := c.beacon.VerifyThreshold(msg, aggregatedSigData.ReconstructedRandomBeaconSig, dkg.GroupKey())
+	if err != nil {
+		return fmt.Errorf("internal error while verifying beacon signature: %w", err)
+	}
+	if !beaconValid {
+		return fmt.Errorf("invalid random beacon signature: %w", signature.ErrInvalidFormat)
+	}
+	// verify the aggregated staking signature next (more costly)
+	// TODO: eventually VerifyMany will be a method of a stateful struct. The struct would
+	// hold the message, all the participants keys, the latest verification aggregated public key,
+	// as well as the latest list of signers (preferably a bit vector, using indices).
+	// VerifyMany would only take the signature and the new list of signers (a bit vector preferably)
+	// as inputs. A new struct needs to be used for each epoch since the list of participants is upadted.
```

Comment:
We can update this TODO as we're choosing a different path.

```suggestion
  // TODO: use new V2 module module/signature.PublicKeyAggregator
```